### PR TITLE
fix(linter/no-null): incorrect fixer for `NullLiteral` within `ReturnStatement`

### LINKED ISSUE #5194: bug(linter) incorrect fixer in `no_null`
see https://github.com/oxc-project/oxlint-ecosystem-ci/actions/runs/10518058484/job/29218160739?pr=25

```ts
return null!
```
is being fixed to 
```ts
return !;
```
which is invalid TS

### PR DESCRIPTION
**Pull request by Dunqing**
close: #5194

It’s a long time no contributed to Linter, I'd like to try it

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#5247** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5247?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/unicorn/no_null.rs [hunk 1]
```diff
@@ -199,7 +199,17 @@ impl Rule for NoNull {
             }
             (AstKind::ReturnStatement(_), _) => {
                 ctx.diagnostic_with_fix(no_null_diagnostic(null_literal.span), |fixer| {
-                    fixer.delete(null_literal)
+                    let mut null_span = null_literal.span;
+                    // Find the last parent that is a TSAsExpression (`null as any`) or TSNonNullExpression (`null!`)
+                    for parent in ctx.nodes().iter_parents(node.id()).skip(1) {
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/unicorn/no_null.rs:204]: What happens if there's a null in a call expression? `foo(null) as any[]`

Comment (DonIsaac) [reply crates/oxc_linter/src/rules/unicorn/no_null.rs:204]: Or `foo(null!)`

Comment (Dunqing) [reply crates/oxc_linter/src/rules/unicorn/no_null.rs:204]: Only `null` in ReturnStatement will be deleted. So:

`foo(null) as any[]` -> `foo(undefined) as any[]`
`foo(null)` -> `foo(undefined)`

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**